src/suggestion-box.ts

```diff
@@ -0,0 +1,150 @@
+import { LitElement,html,css } from "lit";
+import { customElement,property,state } from "lit/decorators.js";
+
+@customElement("aalam-sgn-box")
+export class AalamSuggestionBoxElement extends LitElement {
+    constructor() {
+        super();
+        this.input.addEventListener("keydown",this.handlearrow.bind(this));
+    }
+    static styles = css`
+    :host {
+      display: grid;
+      justify-content:center;
+      align-items:center; 
+      margin:50px;  
+    }}`;
+
+    @property()
+    countheight:number = 0;
+
+    @property()
+    result:any[] = []
+
+    @property()
+    input:any = document.getElementById('inputbox');
+
+    @state()
+    minchar:any = this.input.getAttribute('minchar')
+
+    @state()
+    state:boolean = true;
+
+    @property()
+    currentindex:number = 0;
+
+    @property()
+    listkey:any = [];
+
+    @property({type:Array, attribute: false})
+    list : object[];
+
+    @property({type:String, attribute:false})
+    listKey:string;
+
+    @state()
+    loadmore_index:number = 0;
+
+    render() {
+      return html `
+      <slot name="input" @input=${this.handleInput}></slot>
+      <slot name="list-item" class="list-items">
+      </slot>
+      <ul>
+        ${
+          this.result.map((items,i) => {
+            if(i<this.loadmore_index+5) {
+              return html `
+              <li @click=${this.fillInput} class="itemlist">${ ( typeof items === "object" ) ?  items.suggestion : items}</li>
```

Comment:
We havent mandated the suggestion key to be present in the result if the result is an array of objects. Rather we support a .html attribute if present we give it as a dom, else we parse the sgn-item slot template and substitute the key values in that template.